mam/v2/sponge/spongos_types.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * MAM is based on an original implementation & specification by apmi.bsu.by
+ * [ITSec Lab]
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#ifndef __MAM_V2_SPONGE_SPONGOS_TYPES_H__
+#define __MAM_V2_SPONGE_SPONGOS_TYPES_H__
+
+#include "mam/v2/defs.h"
+#include "mam/v2/sponge/sponge.h"
+#include "mam/v2/trits/trits.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct mam_spongos_s {
+  mam_sponge_t *sponge;
+  size_t pos;
+} mam_spongos_t;
+
+size_t mam_spongos_serialized_size(mam_spongos_t const *const spongos);
+
+void mam_spongos_serialize(mam_spongos_t const *const spongos, trits_t trits);
+
+void mam_spongos_deserialize(trits_t *const trits,
+                             mam_spongos_t *const spongos);
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif  // __MAM_V2_SPONGE_SPONGOS_TYPES_H__
```

Comment:
Do we really need to separate ? Since we don't have a container ?

Reply:
Yes, but for different reason this time,
Spongos depends on pb3 for serialization
And pb3 depends on Spongos too,
so some sort of separation is needed to break circular dependency

Reply:
Alright